File: 2170-count-number-of-maximum-bitwise-or-subsets/2170-count-number-of-maximum-bitwise-or-subsets.py
class Solution(object):
    def countMaxOrSubsets(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ma = 0
        ans = 0

        def f(i, val):
            nonlocal ma, ans
            if i == len(nums):
                if val == ma:
                    ans += 1
                elif val > ma:
                    ma = val
                    ans = 1
                return

            f(i + 1, val | nums[i])
            f(i + 1, val)
        
        f(0, 0)
        return ans
        

File: 2170-count-number-of-maximum-bitwise-or-subsets/2170-count-number-of-maximum-bitwise-or-subsets.cpp
class Solution {
public:
    int countMaxOrSubsets(vector<int>& nums) {
        int ma = 0;
        int c = 0;
        function<void (int, int)> f = [&] (int i, int val) -> void {
            if(i == nums.size()) {
                if(val > ma) {
                    ma = val;
                    c = 1;
                } else if(val == ma) {
                    c++;
                }
                return;
            }
            f(i + 1, val);
            f(i + 1, nums[i] | val);
        };

        f(0, 0);

        return c;
    }
};